### PR TITLE
Fix some more fault injection test failures

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs [hunk 1]
```diff
@@ -27,24 +27,22 @@ public partial class MessageSendFaultInjectionTests : E2EMsTestBase
         public async Task Message_TcpConnectionLossSendRecovery_Amqp()
         {
             await SendMessageRecoveryAsync(
-                    TestDeviceType.Sasl,
-                    new IotHubClientAmqpSettings(),
-                    FaultInjection.FaultType_Tcp,
-                    FaultInjection.FaultCloseReason_Boom,
-                    FaultInjection.DefaultFaultDelay)
-                .ConfigureAwait(false);
+                TestDeviceType.Sasl,
```
Comment (abhipsaMisra) [inline e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs:30]: Tab fix

File: e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs [hunk 2]
```diff
@@ -95,161 +90,131 @@ public async Task Message_TcpConnectionLossSendRecovery_MqttWs()
...
+                new IotHubClientAmqpSettings(IotHubClientTransportProtocol.WebSocket),
+                FaultInjection.FaultType_Throttle,
+                FaultInjection.FaultCloseReason_Boom,
+                FaultInjection.DefaultFaultDelay,
+                FaultInjection.DefaultFaultDuration).ConfigureAwait(false);
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        [ExpectedException(typeof(IotHubThrottledException))]
```
Comment (abhipsaMisra) [inline e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs:193]: Update the test to expect only IotHubThrottledException and remove generic timeout related exceptions as acceptable expected exceptions.

File: e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs [hunk 3]
```diff
@@ -446,8 +400,8 @@ public async Task Message_GracefulShutdownSendRecovery_MqttWs()
             Func<IotHubDeviceClient, TestDevice, Task> testOperation = async (deviceClient, testDevice) =>
             {
                 (Client.Message testMessage, string payload, string p1Value) = MessageSendE2ETests.ComposeD2cTestMessage(Logger);
-                using var cts = new CancellationTokenSource(retryDuration);
-                await deviceClient.SendEventAsync(testMessage).ConfigureAwait(false);
+                using var cts = new CancellationTokenSource(operationTimeoutInMilliSecs);
+                await deviceClient.SendEventAsync(testMessage, cts.Token).ConfigureAwait(false);
```
Comment (abhipsaMisra) [inline e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs:404]: The operation timeout evaluated above wasn't being passed on to the API call.

File: e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs [hunk 4]
```diff
@@ -95,186 +90,131 @@ public async Task Message_TcpConnectionLossSendRecovery_MqttWs()
...
-            catch (IotHubThrottledException) { }
-            catch (IotHubCommunicationException ex)
-            {
-                Assert.IsInstanceOfType(ex.InnerException, typeof(OperationCanceledException));
-            }
-            catch (TimeoutException) { }
-        }
-
-        [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
-        public async Task Message_ThrottledConnectionLongTimeNoRecovery_Http()
```
Comment (abhipsaMisra) [inline e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs:256]: We're on track to remove HTTP support for v2 device clients, so removing this test. This test also behaves differently than expected - service disconnects immediately after receiving this fault request, but the disconnection is an ungraceful I/O disconnect (and not a throttling/quota exceeded specific error code). Our [tests](https://github.com/Azure/azure-iot-sdk-csharp/blob/8e5c5bd102c23ddb94175823f219c4514831e0e2/e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs#L307-L328) also seem to accept TimeoutException and OperationCancelledException as acceptable exception types, which makes these tests non-deterministic in nature.

**Review by timtay-microsoft — APPROVED** [review]

File: e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs (hunk 1 above)
Comment (drwill-ms) [reply e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs:30]: That tabbing was intentional.

**Pull request merged**